apps/user-ms/src/main/java/it/pagopa/selfcare/user/service/UserNotificationServiceImpl.java

```diff
@@ -104,14 +139,19 @@ private Map<String, String> buildEmailDataModel(UserInstitution institution, Pro
         return dataModel;
     }
 
-    private Uni<Void> buildAndSendEmailNotification(org.openapi.quarkus.user_registry_json.model.UserResource user, UserInstitution institution, Product product, String templateName, String subject, String loggedUserName, String loggedUserSurname) {
+    private Uni<Void> buildDataModelAndSendEmail(org.openapi.quarkus.user_registry_json.model.UserResource user, UserInstitution institution, Product product, String templateName, String subject, String loggedUserName, String loggedUserSurname) {
         return Uni.createFrom().voidItem().onItem().transformToUni(Unchecked.function(x -> {
             String email = retrieveMail(user, institution);
             Map<String, String> dataModel = buildEmailDataModel(institution, product, loggedUserName, loggedUserSurname);
             return Uni.createFrom().item(getContent(templateName, dataModel))
-                    .onItem().transformToUni(content ->  mailService.sendMail(email, content.toString(), subject));
+                    .onItem().transformToUni(content -> this.sendEmailNotification(templateName, subject, email, dataModel));
         }));
     }
+    private Uni<Void> sendEmailNotification(String templateName, String subject, String email, Map<String, String> dataModel) {
+        return Uni.createFrom().voidItem().onItem().transformToUni(Unchecked.function(x -> Uni.createFrom()
+                .item(getContent(templateName, dataModel))
+                .onItem().transformToUni(content -> mailService.sendMail(email, content.toString(), subject))));
```

Comment:
pay attention to `getContent` method that can block main thread, look at [this](https://smallrye.io/smallrye-mutiny/1.6.0/guides/imperative-to-reactive/)

You could refactor in this way:

```
Uni.createFrom().item(() -> getContent(templateName, dataModel))
                .runSubscriptionOn(Infrastructure.getDefaultWorkerPool())
                .onItem().transformToUni(content -> mailService.sendMail(email, content.toString(), subject)))
```